File: AnYang/Greedy/455.py
# 2019-07-13 Sat

# 455. Assign Cookies

# 先占个坑，借鉴dalwise的解法
# 用两个指针分别两个排好序的list
class Solution:
    def findContentChildren(self, g, s):
        g.sort()
        s.sort()
        i = 0
        for e in s:
            if i == len(g):
                break
            if e >= g[i]:
                i += 1
        return i

File: 小脑袋/455.cpp
//贪心。用最小的满足孩子
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int i = 0, j = 0;
        while (i < g.size() && j < s.size()){
            if (g[i] <= s[j])
                i++;
            j++;
        }
        return i;
    }
};
